### ISSUE TITLE
Set *local* display name for a room

### ISSUE THREAD

**Issue by monnier** [vector-im/riot-android#1564]
[ Using the Riot app from F-Droid. ]

In "people" rooms, I can't seem to find a way to name the room, without that name also being imposed on the other participant.

For example, say I have a "people room" (I think it's also called a "direct chat" elsewhere) to talk to my friend John, I'd naturally want to name it "John", but then he'll find this name makes no sense for him, who'd rather name it "Stef".

Am I missing something or does Riot lack this feature?

**Comment by ylecollen** [comment]
@monnier 
You can set a room name (open the room and tap on it) but it requires some power levels.
Either you created the room or you could ask to the other members to rename the room or give the power levels to manage it.
But, the room name will be the same for any user.

**Comment by monnier** [comment]
OK, so I did not miss anything: Riot lacks the ability to set a *local* display name, one that only affects my view of the room, and not that of other participants.

Please consider this as a feature request, then.  Ideally, this local display name should be shared by all the clients I use (i.e. should be stored along with my subscriptions), but it'd be acceptable to instead store it in the riot client, instead (so I'd have to set it anew for every client I use to access my account).

**Comment by nvbln** [comment]
I might be wrong but I get the idea that there is a bit of confusion here. Although Riot does indeed lack the ability to set a local display name, what you want is to see the room name as the name of the person you're talking to, if I'm not mistaken?

This is the default behaviour for all rooms that are tagged as 'direct-chat.' As far as I know, if you have a direct chat room with a custom title, dropping the title altogether should cause the room title to show the name of the other participant. Which should satisfy your request right?

**Comment by monnier** [comment]
Yes and no: the direct-chat room may have name `Frederic McCarthy` (because that's the name of the other participant) but I'd rather call it `Fred`.  I guess rather than "local display name" you could call it "nickname".  I mostly need it for direct-chat rooms, but having it for other rooms wouldn't hurt (e.g. I might prefer the local name `Emacs` for the room `Emacs en Español` if that's the only Emacs-related room in which I participate).

**Comment by nvbln** [comment]
As this would probably be a feature that affects all Riot clients you might want to suggest it at [riot-meta](https://github.com/vector-im/riot-meta/issues).

**Comment by krombel** [comment]
I think it is related to matrix-org/matrix-doc#690
